exporter/elasticexporter/internal/translator/elastic/traces.go

```diff
@@ -416,16 +456,19 @@ func (c *transactionContext) setHTTPStatusCode(statusCode int) {
 }
 
 type spanContext struct {
-	model   model.SpanContext
-	http    model.HTTPSpanContext
-	httpURL url.URL
-	db      model.DatabaseSpanContext
+	model              model.SpanContext
+	http               model.HTTPSpanContext
+	httpURL            url.URL
+	db                 model.DatabaseSpanContext
+	destination        model.DestinationSpanContext
+	destinationService model.DestinationServiceSpanContext
```

Comment:
Why are you introducing `destination` and `destinationService` in this struct, couldn't you directly use `c.model.Destination`  in `setDestinationAddress`  and `setDestinationService`?

Reply:
Fewer allocations. `destination` and `destinationService` are are part of the same heap object that will end up pointing to them (spanContext).

Reply:
that makes sense, thanks; didn't pay attention that `model.Destination`  and `model.DestinationService` are actually pointers.